Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -747,7 +801,7 @@ public class DependencyResolver<
         //
         // FIXME: We should have a test for this, probably by adding some kind
         // of statistics on the number of backtracks.
-        for constraint in constraints {
+        for constraint in constraints where constraint.requirement != .unversioned {
```

Comment:
Actually that is incorrect because we might discover containers as we resolve. That means unless we already know  every container that is going to be encountered, this will be wrong. I think the reverse of this might work i.e. you pass in the names of container you want to exclude from resolution.